### PR TITLE
Fixes #24268 - Add permissions to canned admin

### PR DESCRIPTION
**Pull request by xprazak2**
Depends on https://github.com/theforeman/foreman/pull/5841

### CODE REVIEW

**Comment by chris1984** [comment]
[test katello]

**Comment by xprazak2** [comment]
Tests will be failing until the foreman part is merged.

**Comment by san7ket** [comment]
I am getting a error, on rake:compile while I try to compile it with foreman's PR. 
Looks like we need a rebase here on katello 3.9.0
```
rake aborted!
Foreman::Exception: ERF42-2432 [Foreman::Exception]: Invalid role name, only ''Viewer', 'Manager', 'Organization admin', 'System admin'' is allowed to be extended from a plugin
/home/vagrant/foreman/app/registries/foreman/plugin/rbac_support.rb:42:in `check_role_name_before_extending'
/home/vagrant/foreman/app/registries/foreman/plugin/rbac_support.rb:33:in `block in add_permissions_to_default_roles'
/home/vagrant/foreman/app/registries/foreman/plugin/rbac_support.rb:32:in `each'
/home/vagrant/foreman/app/registries/foreman/plugin/rbac_support.rb:32:in `add_permissions_to_default_roles'
/home/vagrant/foreman/app/registries/foreman/plugin.rb:310:in `block (2 levels) in add_permissions_to_default_roles'
```

File: lib/katello/plugin.rb [hunk 1]
```diff
@@ -300,6 +300,8 @@
   Katello::PermissionCreator.new(self).define
   add_all_permissions_to_default_roles
 
+  add_permissions_to_default_roles 'Canned admin' => [:create_lifecycle_environments, :create_content_views]
```
Comment (san7ket) [inline lib/katello/plugin.rb]: Needs a name change here to System admin

**Comment by xprazak2** [comment]
Rebased and updated.

**Comment by johnsonm325** [comment]
[test katello]

**Comment by johnsonm325** [comment]
@xprazak2 Looks like we have another "rake aborted!" error in the tests.
```
rake aborted!
ActiveRecord::NoDatabaseError: FATAL:  database "katello-pr-test-3452-development" does not exist
```

**Comment by xprazak2** [comment]
[test katello]

**Comment by xprazak2** [comment]
@johnsonm325, what do I have to do to make the tests pass? The db still fails to get configured properly...

**Comment by johnsonm325** [comment]
[test katello]

File: lib/katello/plugin.rb [hunk 2]
```diff
@@ -315,6 +315,8 @@
   Katello::PermissionCreator.new(self).define
   add_all_permissions_to_default_roles
 
+  add_permissions_to_default_roles 'System admin' => [:create_lifecycle_environments, :create_content_views]
```
Comment (evgeni) [inline lib/katello/plugin.rb]: I *think* `add_permissions_to_default_roles` needs to go *inside* `Foreman::Plugin.register :katello do` to work properly

Comment (evgeni) [reply lib/katello/plugin.rb]: nvm, I can't count.

Comment (xprazak2) [reply lib/katello/plugin.rb]: The whole file is one big block, I think it is inside.

**Comment by evgeni** [comment]
Opened https://projects.theforeman.org/issues/25312

**Comment by tbrisker** [comment]
core side has been merged

**Comment by evgeni** [comment]
[test katello]

**Comment by evgeni** [comment]
Tests pass now. Rubocop is unhappy, but that's unrelated to this PR.

**Comment by beav** [comment]
is this no longer "blocked by another pr"?

**Comment by evgeni** [comment]
Correct, both the core feature and the test fix were merged

**Comment by san7ket** [comment]
Yes, the core feature is merged, I have one question though, if this is merged now,  will we still get this in katello 3.9 RC (if there's another) or it goes in 3.10 ?

**Comment by jturel** [comment]
Looks good. I pinged @zjhuntin on getting it into 3.9 since he is the release owner.

One question - how was the determination made as to which permissions should be applied to 'System admin'

**Comment by xprazak2** [comment]
'System admin' is supposed to create new organizations and because the organization default content view and library lifecycle env are created during that process, permissions for these two are needed for organization to be successfully created.

**Comment by jturel** [comment]
@san7ket this will be part of 3.9!

**Comment by xprazak2** [comment]
I opened https://github.com/Katello/katello/pull/7826 with a cp into 3.9 branch.

**Pull request merged**